definitions/features/tar.rb

```diff
@@ -0,0 +1,74 @@
+class Features::Tar < ForemanMaintain::Feature
+  metadata do
+    label :tar
+    preparation_steps { Procedures::Packages::Install.new(:packages => %w[tar]) }
+  end
+
+  # Run system tar
+  # @param [Hash] options tar configuration
+  # @option options [String] :archive archive name
+  # @option options [String] :command ('create') tar operation command
+  # @option options [Array] :exclude dirs or files to exclude
+  # @option options [String] :listed_incremental .snar file to do incremental backup
+  # @option options [String] :transform sed expression to transform the filenames
+  # @option options [String] :volume_size size of tar volume (will try to split the archive when set)
+  # @option options [String] :files (*) files to operate on
+  def run(options = {})
+    volume_size = options.fetch(:volume_size, nil)
+    validate_volume_size(volume_size) unless volume_size.nil?
+
+    tar_command = ['tar']
+    tar_command << '--selinux'
+    tar_command << "--#{options.fetch(:command, 'create')}"
+    tar_command << "--file=#{options.fetch(:archive)}"
+
+    if volume_size
+      split_tar_script = default_split_tar_script
+      tar_command << "--tape-length=#{volume_size}"
+      tar_command << "--new-volume-script=#{split_tar_script}"
+    end
```

Comment:
I had to separate these options like so:

```
diff --git a/definitions/features/tar.rb b/definitions/features/tar.rb
index 5dfa8ee..6bf24d4 100644
--- a/definitions/features/tar.rb
+++ b/definitions/features/tar.rb
@@ -13,6 +13,7 @@ class Features::Tar < ForemanMaintain::Feature
   # @option options [String] :transform sed expression to transform the filenames
   # @option options [String] :volume_size size of tar volume (will try to split the archive when set)
   # @option options [String] :directory change to directory DIR
+  # @option options [Boolean] :split_data splits tar with rotating volume script
   # @option options [Boolean] :multi_volume create/list/extract multi-volume archive
   # @option options [Boolean] :overwrite overwrite existing files when extracting
   # @option options [Boolean] :gzip filter the archive through gzip
@@ -27,10 +28,10 @@ class Features::Tar < ForemanMaintain::Feature
     tar_command << '--gzip' if options[:gzip]
     tar_command << "--#{options.fetch(:command, 'create')}"
     tar_command << "--file=#{options.fetch(:archive)}"
+    tar_command << "--tape-length=#{volume_size}" if volume_size
 
-    if volume_size
+    if options[:split_data]
       split_tar_script = default_split_tar_script
-      tar_command << "--tape-length=#{volume_size}"
       tar_command << "--new-volume-script=#{split_tar_script}"
     end
 
@@ -82,5 +83,4 @@ class Features::Tar < ForemanMaintain::Feature
 
     split_tar_script
   end
```

since katello-restore does not use tape-length, but uses --new-volume script
https://github.com/theforeman/foreman-packaging/blob/rpm/develop/packages/katello/katello/restore.rb#L152

Reply:
actually nvm, I think volume_size defaults to nil like you have it, this should be fine